community/detectors/kubernetes_rce_via_open_access/src/main/java/com/google/tsunami/plugins/detectors/rce/kubernetes/RCEInKubernetesClusterWithOpenAccessDetector.java

```diff
@@ -0,0 +1,290 @@
+    // callback-enabled payload.
+    Payload payload = this.payloadGenerator.generate(config);
+    String commandToInject = payload.getPayload();
+    String reqPayload = String.format(payloadFormatString, commandToInject);
+    boolean isPodCreated = false;
+    boolean isVulnerable = false;
+
+    // Create a new Kubernetes Pod with RCE payload in container command args
+    isPodCreated = createPod(networkService, RCE_POD_NAME, reqPayload);
+    if (!isPodCreated) {
+      logger.atInfo().log("Failed to create a pod. Not vulnerable.");
+      return false;
+    }
+
+    // Created a pod, appears vulnerable. Use callback for RCE confirmation if enabled
+    if (!payload.getPayloadAttributes().getUsesCallbackServer()) {
+      logger.atWarning().log("Pod created, appears vulnerable, but callback server is disabled.");
+      // NOTE: Set to false if only executed payloads/confirmed RCE cases should be reported
+      isVulnerable = true;
+
+    } else {
+      // If there is an RCE, the execution isn't immediate
+      logger.atInfo().log("Waiting for RCE callback.");
+      try {
+        Thread.sleep(10000);
+      } catch (InterruptedException e) {
+        logger.atWarning().withCause(e).log("Failed to wait for RCE result");
+      }
+      // Check if callback server received the request
+      isVulnerable = payload.checkIfExecuted();
+    }
+
+    // Cleanup by removing the created pod
+    deletePod(networkService, RCE_POD_NAME);
+
+    return isVulnerable;
+  }
+
+  private static String buildTargetUrl(NetworkService networkService, String path) {
+    StringBuilder targetUrlBuilder = new StringBuilder();
+    if (NetworkServiceUtils.isWebService(networkService)) {
+
+      targetUrlBuilder.append(NetworkServiceUtils.buildWebApplicationRootUrl(networkService));
+    } else {
+      // Default to HTTPS protocol when the scanner cannot identify the actual service.
+      // which is what Kubernetes API server normally uses.
+      targetUrlBuilder
+          .append("https://")
+          .append(toUriAuthority(networkService.getNetworkEndpoint()))
+          .append("/");
+    }
+    targetUrlBuilder.append(path);
+    return targetUrlBuilder.toString();
+  }
+
+  private boolean createPod(NetworkService networkService, String podName, String payload) {
+    String targetUri = buildTargetUrl(networkService, "api/v1/namespaces/default/pods");
+    logger.atInfo().log("Sending a request to Kubernetes service '%s' to trigger RCE", targetUri);
+
+    HttpRequest req =
+        HttpRequest.post(targetUri)
+            .setHeaders(HttpHeaders.builder().addHeader(CONTENT_TYPE, "application/json").build())
+            .setRequestBody(ByteString.copyFromUtf8(payload))
+            .build();
+
+    try {
+      HttpResponse response = this.httpClient.send(req, networkService);
+      if (response.status().code() == HttpStatus.CREATED.code()
+          && response.bodyString().map(body -> body.contains("tsunami-rce-pod")).orElse(false)) {
```

Comment:
To make it easier to triage filed bugs, let's add the pod creation response into the detection report, in its AdditionalDetails field like https://github.com/google/tsunami-security-scanner-plugins/blob/4373ef8b8f41b920e99acc96ee9f862c77f21403/community/detectors/influxdb_cve_2019_20933/src/main/java/com/google/tsunami/plugins/detectors/cves/cve201920933/Cve201920933VulnDetector.java#L225

Reply:
Ok, I've added additional details to the report as suggested.